packages/environment-check/src/output/markdown.ts

```diff
@@ -259,11 +296,15 @@ function writeMessages(writer: MarkdownWriter, messages: ResultMessage[] = []):
 export function convertResultsToMarkdown(results: EnvironmentCheckResult): string {
     const writer = getMarkdownWriter();
 
-    writer.addH1(t('markdownText.envCheckTitle'));
-    writeEnvironment(writer, results.environment);
-    writeDestinationResults(writer, results.destinationResults, results.destinations);
-    writeDestinations(writer, results.destinations);
-    writeMessages(writer, results.messages);
+    writer.addH1(results.markdownTitle);
+    if (results.toolsExtensions) {
```

Comment:
Can we avoid having an `if` here and output whatever is there in the `EnvironmentCheckResult`. We might fill certain properties only in certain cases, but the writer should just traverse over the results and output markdown, not aware of other logic. This keeps the writer simple. I assume at some point we might also check certain module versions in BAS and destinations in VSCode.